### PR TITLE
chore: Update list of ocaml lsp versions

### PR DESCRIPTION
**Pull request by 3Rafal**

### CODE REVIEW

File: src/ocaml_lsp.ml [hunk 1]
```diff
@@ -112,12 +112,16 @@ let lsp_versions =
          ; "1.14.2"
          ; "1.15.0-4.14"
          ; "1.15.1-4.14"
+         ; "1.16.0"
```
Comment (rgrinberg) [inline src/ocaml_lsp.ml:115]: Should be 1.16.0-4.14 actually

File: src/ocaml_lsp.ml [hunk 2]
```diff
@@ -112,12 +112,16 @@ let lsp_versions =
...
+         ; "1.16.1"
         |] )
     ; ( (5, 0)
       , [| "1.13.2~5.0preview"
          ; "1.15.0~5.0preview1"
          ; "1.15.0-5.0"
          ; "1.15.1-5.0"
+         ; "1.16.0"
```
Comment (rgrinberg) [inline src/ocaml_lsp.ml:123]: There's no 1.16.0 for 5.0

**Comment by rgrinberg** [comment]
How come you've removed 1.16.1 and 1.16.0-4.14? It doesn't really matter because not many people are using them, but there's no harm in keeping the list of versions exhaustive.

**Comment by 3Rafal** [comment]
I brought them back immediately, because I figured out that they may be useful

**Comment by 3Rafal** [comment]
@rgrinberg , ocaml-lsp 1.16.2 was merged to opam-repo: https://github.com/ocaml/opam-repository/pull/23985
This means that we can merge this PR too, right?

**Comment by rgrinberg** [comment]
Yup. Note that we could merge this PR even earlier. We just need to wait until opam-repository is updated before releasing.

**Review by smorimoto — APPROVED** [review]

**Comment by 3Rafal** [comment]
I was thinking of some edge cases, where opam-repository PR would be rejected and then we would need to revert, but the risk is low and even if it happened it probably wouldn't be a big deal. :)

**Comment by voodoos** [comment]
Sorry, I am a little late to the party, but `1.16.1-2` releases also support OCaml 5.1 (alpha2 and very probably the future final release), are we waiting for the release to add it to the list ?

**Comment by rgrinberg** [comment]
Don't think we released yet so you can send a PR for 5.1

**Pull request merged**